pkg/cvo/sync_worker.go

```diff
@@ -818,6 +840,10 @@ func (w *SyncWorker) apply(ctx context.Context, work *SyncWork, maxWorkers int,
 					klog.V(2).Infof("Skipping precreation of %s as unmanaged", task)
 					continue
 				}
+				if err := capability.ResourceEnabled(resourceCapabilities[i], work.Capabilities.EnabledCapabilities); err != nil {
```

Comment:
I'm a bit confused by this.  Is it "include the manifest if the capabilities that that manifest needs match the capabilities we pulled off of the manifest when we loaded it"?  Why can't we pass in the cluster's current set of enabled capabilities, without needing to `[i]` index anything?